fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/control/MigrateV0027LogicalResourceIdent.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * (C) Copyright IBM Corp. 2021
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.fhir.schema.control;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+
+import com.ibm.fhir.database.utils.api.IDatabaseStatement;
+import com.ibm.fhir.database.utils.api.IDatabaseTranslator;
+import com.ibm.fhir.database.utils.common.DataDefinitionUtil;
+
+/**
+ * Populate LOGICAL_RESOURCE_IDENT with records from LOGICAL_RESOURCES
+ */
+public class MigrateV0027LogicalResourceIdent implements IDatabaseStatement {
+
+    // The FHIR data schema
+    private final String schemaName;
+
+    /**
+     * Public constructor
+     * @param schemaName
+     */
+    public MigrateV0027LogicalResourceIdent(String schemaName) {
+        this.schemaName = schemaName;
+    }
+
+    @Override
+    public void run(IDatabaseTranslator translator, Connection c) {
+        final String logicalResources = DataDefinitionUtil.getQualifiedName(schemaName, "LOGICAL_RESOURCES");
+        final String logicalResourceIdent = DataDefinitionUtil.getQualifiedName(schemaName, "LOGICAL_RESOURCE_IDENT");
+        final String DML = ""
+                + "INSERT INTO " + logicalResourceIdent +"(resource_type_id, logical_id, logical_resource_id) "
+                + "     SELECT resource_type_id, logical_id, logical_resource_id "
+                + "       FROM " + logicalResources;
```

Comment:
will citus have trouble with this since logical_resources is distributed (and by a different key than logicalResourceIdent)?

Reply:
logical_resource_ident is the one distributed table which is distributed by a different column, which in this case is logical_id.

Reply:
OK. I think I was just worried about the performance of the statement given that it will necessarilly be copying a bunch of data across a bunch of different nodes.
I talked to Robin offline and he reminded me that migration from single-node postgres to multi-node citus is not supported.
So the migration into logical_resource_ident should never be needed on Citus, because with Citus we'll start out with the V0027 schema.